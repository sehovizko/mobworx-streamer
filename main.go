package main

import (
	"hls.streaming.com/streaming"
)

func main() {
	streaming.InitConfig()
}
