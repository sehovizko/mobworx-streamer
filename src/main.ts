import { App, Stack, StackProps } from "aws-cdk-lib";
import { Construct } from "constructs";
import { EndpointNestedStack } from "./endpoint/endpoint.nested-stack";
import { VpcNestedStack } from "./vpc.nested-stack";
import { StreamingNestedStack } from "./streaming/streaming.nested-stack";

export class StreamerStack extends Stack {
  constructor(scope: Construct, id: string, props: StackProps = {}) {
    super(scope, id, props);
    const { vpc } = new VpcNestedStack(this, "VPC");
    new EndpointNestedStack(this, "EndpointNestedStack", { vpc });
    new StreamingNestedStack(this, "StreamingNestedStack", { vpc });
  }
}

// for development, use account/region from cdk cli
const devEnv = {
  account: process.env.CDK_DEFAULT_ACCOUNT,
  region: process.env.CDK_DEFAULT_REGION,
};

const app = new App();

new StreamerStack(app, "mobworx-streamer-dev", { env: devEnv });
// new MyStack(app, 'mobworx-streamer-prod', { env: prodEnv });

app.synth();
