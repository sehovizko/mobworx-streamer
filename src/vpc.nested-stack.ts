import { NestedStack } from "aws-cdk-lib";
import { Vpc } from "aws-cdk-lib/aws-ec2";

export class VpcNestedStack extends NestedStack {
  vpc = new Vpc(this, "VPC");
}
