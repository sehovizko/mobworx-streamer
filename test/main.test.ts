import { App } from "aws-cdk-lib";
import { Template } from "aws-cdk-lib/assertions";
import { StreamerStack } from "../src/main";

test("Snapshot", () => {
  const app = new App();
  const stack = new StreamerStack(app, "test");

  const template = Template.fromStack(stack);
  expect(template.toJSON()).toMatchSnapshot();
});
