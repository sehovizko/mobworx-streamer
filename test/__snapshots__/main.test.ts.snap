// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot 1`] = `
{
  "Parameters": {
    "BootstrapVersion": {
      "Default": "/cdk-bootstrap/hnb659fds/version",
      "Description": "Version of the CDK Bootstrap resources in this environment, automatically retrieved from SSM Parameter Store. [cdk:skip]",
      "Type": "AWS::SSM::Parameter::Value<String>",
    },
  },
  "Resources": {
    "EndpointNestedStackNestedStackEndpointNestedStackNestedStackResource3F1846A3": {
      "DeletionPolicy": "Delete",
      "Properties": {
        "TemplateURL": {
          "Fn::Join": [
            "",
            [
              "https://s3.",
              {
                "Ref": "AWS::Region",
              },
              ".",
              {
                "Ref": "AWS::URLSuffix",
              },
              "/",
              {
                "Fn::Sub": "cdk-hnb659fds-assets-\${AWS::AccountId}-\${AWS::Region}",
              },
              "/9988f6773d5dae3a4081ef39a6f611c68d328c0596bbf0bcb3e63a278abd41f2.json",
            ],
          ],
        },
      },
      "Type": "AWS::CloudFormation::Stack",
      "UpdateReplacePolicy": "Delete",
    },
  },
  "Rules": {
    "CheckBootstrapVersion": {
      "Assertions": [
        {
          "Assert": {
            "Fn::Not": [
              {
                "Fn::Contains": [
                  [
                    "1",
                    "2",
                    "3",
                    "4",
                    "5",
                  ],
                  {
                    "Ref": "BootstrapVersion",
                  },
                ],
              },
            ],
          },
          "AssertDescription": "CDK bootstrap stack version 6 required. Please run 'cdk bootstrap' with a recent version of the CDK CLI.",
        },
      ],
    },
  },
}
`;
